### ISSUE TITLE
doc: how to deploy Cilium Operator on a Local OpenShift cluster

### ISSUE THREAD

**Issue by elmazzun** [cilium/cilium#27973]
Hi!

# TL;DR

After running Cilium on KinD, I'd like to try it on my virtualized OpenShift cluster [Local OpenShift](https://developers.redhat.com/products/openshift-local/overview) but it fails to deploy the Cilium DaemonSet because of some restrictive permissions.

Is there a _documented way_ to deploy Cilium on a OpenShift cluster?

# My problem in detail

The Cilium Pod Operator is running fine, thus I proceed to create a CiliumConfig resource with default values:
```yaml
apiVersion: cilium.io/v1alpha1
kind: CiliumConfig
metadata:
  name: cilium-openshift-default
  namespace: cilium
spec: {}
```
Then the DaemonSet `cilium` is created, which already fails with the following error:
```
Error creating: pods "cilium-" is forbidden: unable to validate against any security context constraint: 
[provider "anyuid": Forbidden: not usable by user or serviceaccount, provider restricted-v2: .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, 
spec.volumes[1]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[2]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[3]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[4]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[5]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[6]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[7]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[8]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[10]: Invalid value: "hostPath": hostPath volumes are not allowed to be used, 
spec.volumes[11]: Invalid value: "hostPath": hostPath volumes are not allowed to be used,
spec.initContainers[0].securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used, 
spec.initContainers[1].securityContext.seLinuxOptions.level: Invalid value: "s0": must be s0:c26,c5, 
spec.initContainers[1].securityContext.seLinuxOptions.type: Invalid value: "spc_t": must be , 
spec.initContainers[1].securityContext.capabilities.add: Invalid value: "SYS_ADMIN": capability may not be added,
```

I fixed this  by running `oc adm policy add-scc-to-user privileged -z cilium`: this made the DaemonSet deploy the only Pod (Local OpenShift is a single-Node cluster, with a master/worker Node).

Now DaemonSet Pods starts but the problem is now:
```
cp: cannot create regular file '/hostbin/cilium-mount': Read-only file system
```

# Environment

## Cilium
Operator Cilium 1.14.1+xdd859b7 provided by Isovalent

## Local OpenShift
```bash
$ crc version
CRC version: 2.19.0+a71226
OpenShift version: 4.12.13
Podman version: 4.4.4
```

**Comment by youngnick** [comment]
Do the standard install docs at https://docs.cilium.io/en/stable/installation/k8s-install-openshift-okd/ not apply because it's Local Openshift?

In any case, the Cilium agent definitely expects to be able to create a read-write hostPath volume, so if the nodes have read-only filesystems only, then that may be a problem.

**Comment by youngnick** [comment]
After asking around however, it seems that the OSS Cilium community no longer maintains OLMs - you can contact one of the vendors listed at https://github.com/cilium/cilium-olm#cilium-olm for someone who can help you with that.

**Comment by elmazzun** [comment]
I just realized the typo in the title I wrote, I swear I did it by mistake
Thanks @youngnick I'll do as you suggested

**Comment by domq** [comment]
Today I found out that the same issue arises on a “regular” (not local) cluster. I'm trying to follow [this Isovalent blog post](https://isovalent.com/blog/post/cilium-hubble-observability-cni-chaining-mode/) on a freshly-installed OpenShift 4.15.5, and I'm getting the exact same error message:

```
oc -n kube-system logs cilium-j4rgp -c mount-cgroup
cp: cannot create regular file '/hostbin/cilium-mount': Read-only file system  
```

The reason why the “official” instructions for Cilium on OpenShift don't apply in my case, is that I want to try out [CNI chaining](https://isovalent.com/blog/post/cilium-hubble-observability-cni-chaining-mode/) i.e. I don't want to change the install-time CNI (if I can avoid it).

**Comment by benzaidfoued** [comment]
hey @domq ,

I've same issue on OCP 4.14/4.15, i gave cilium SA all permissions needed on host level but that did not fixed the issue , could you please share more details on your end , was that issue fixed for you ?
